rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -431,16 +509,63 @@ def _build_tf_train_graph(self) -> Tuple["tf.Tensor", "tf.Tensor"]:
             self.scale_loss,
         )
 
-    def _build_tf_pred_graph(
-        self, session_data: "train_utils.SessionData"
-    ) -> "tf.Tensor":
+    def combine_sparse_dense_features(
+        self,
+        batch: Dict[Text, tf.Tensor],
+        key_prefix: Text,
+        feature_dim: int,
+        batch_size: int,
+    ):
+        key_sparse = f"{key_prefix}sparse"
+        key_dense = f"{key_prefix}dense"
+
+        # apply mean/sum to convert sequence to sentence features
+
+        if key_dense in batch and key_sparse in batch:
+            _sparse = tf.math.reduce_sum(
+                train_utils.tf_dense_layer(
+                    batch[key_sparse],
+                    feature_dim,  # TODO define proper size
+                    "a",
+                    self.C2,
+                    feature_dim=feature_dim,
+                    batch_size=batch_size,
+                )
+            )
+            _dense = tf.math.reduce_mean(batch[key_dense], axis=1)
+
+            return tf.concat([_sparse, _dense])
+
+        if key_dense in batch:
+            return tf.math.reduce_mean(batch[key_dense], axis=1)
+
+        if key_sparse in batch:
+            return tf.math.reduce_sum(
+                train_utils.tf_dense_layer(
```

Comment:
sorry for moving ideas back and force, now seeing it implemented, I don't think we need to wrap sparse_dense and normal dense into one function. This is the only place where we need it, so maybe it is clearer to call sparse_dense directly here